# Problem of the Week 1

Name: Adarsh Mallya

Student ID: 017086225

Date: August 21, 2025

**Motivation**

This week's Problem of the Week is designed to get us organized and to install the Python packages and tools we will use in this course. Run the following cells (after updating the first one) to showcase your directories, your conda environment, and your installed packages.

After your cells run successfully, generate a PDF of this notebook and turn both your PDF and your notebook in on Canvas.

## Problem 1.1: Course file structure

I have created a file structure for this course as follows:

In [19]:
import os

course_directory = '/Users/adarsh/Desktop/CS_171/'
                  # ^ Enter the location of your directory here
                  #   This should be a folder dedicated to CS 171
print('My course directory is: ')
print(course_directory)
print(' ')

print('The subdirectories in my course directory are: ') # should include Homework, Lecture, 
                                                         # Project, Problem of the Week
for file_name in os.listdir(course_directory):
    d = os.path.join(course_directory, file_name)
    if os.path.isdir(d): 
        print(' - '+file_name)
print(' ')

print('Currently, I am working in the following directory: ')
print(os.getcwd())

My course directory is: 
/Users/adarsh/Desktop/CS_171/
 
The subdirectories in my course directory are: 
 - Lecture
 - Project
 - Problem_Of_The_Week
 - Homework
 
Currently, I am working in the following directory: 
/Users/adarsh/Desktop/CS_171


## Problem 1.2: `conda` Environment

For this course, I created a `conda` environment called `cs171`. You can observe my environment as follows:

In [20]:
print('My current Python environment is: '+os.environ['VIRTUAL_ENV'])
# I didn't use conda as my package manager

My current Python environment is: /Users/adarsh/.pyenv/versions/3.12.9/envs/CS171


This conda environment was install with version 3.12, which we can verify here:

In [21]:
import sys
version_info = sys.version_info
print('My python version is: '+str(version_info[0])+'.'+str(version_info[1]))

My python version is: 3.12


## Problem 1.3: Package Installation

I have installed all of the pertinent packages we will use in this class in my conda environment. I can verify by importing the following packages and ensuring there are no errors:

In [22]:
import numpy
import matplotlib
import torch
import torchvision
import pandas
import netCDF4
import scipy
import sklearn

In short, I am organized and set to dive into Machine Learning!